src/Model/CarParking.php

```diff
@@ -3,7 +3,11 @@
 
 namespace App\Model;
 
-class CarParking implements ParkingInterface
+class CarParking extends AbstractParking implements ParkingInterface
 {
-
+    public function canPark(VehicleInterface $vehicle): bool
+    {
+        $result = (($vehicle instanceof EngineVehicleInterface) && (!$vehicle instanceof Bus)) ?? false;
+            return $result;
```

Comment:
Warunek bardzo nieczytelny. Dodatkowo `??`

Rozbij to używając podejścia Fail Fast